### PR TITLE
feat(builder): add `--metafile` flag

### LINKED ISSUE #6211: [Feature] add builder option to produces `esbuild`'s `metafile`
- [x] I'd be willing to implement this feature ([contributing guide](https://yarnpkg.com/advanced/contributing))
- [x] This feature is important to have in this repository; a contrib plugin wouldn't do

**Describe the user story**

As a plugin developer who uses `@yarnpkg/builder@^4` to build the plugin I wrote, I want to have access to [`eslint`'s  metafile](https://esbuild.github.io/api/#metafile), so that I can use it in additional tooling and analysis.

**Describe the solution you'd like**

Have a CLI switch, that enables [`eslint`'s  metafile](https://esbuild.github.io/api/#metafile).
Something along `builder build plugin --metafile`.
The CLI switch would be "disabled" by default, meaning no file is written.
If the CLI switch is set to "true", then `esbuild`'s metafile is written to `bundles/${name}.meta.json`.

**Describe the drawbacks of your solution**

This described solution would add a new optional CLI switch. If not used, it would not change any o current behaviors,
There should be no drawbacks with that, except that it keeps the current "vendor lock-in" with `esbuild`

**Describe alternatives you've considered**

I thought about having an option to pass basic `esbuild` config into the build command, which then is merged/overwritten by `yarn` specific settings, and the result is applied to `esbuld`.
The settings could be stored in the `yarnrc` file in a section called `builder-build` with a subsection per purpose (`plugin` or `bundle`). 
This would open up a lot of custom options, true, but it would require people to learn the external structure and documentation and the whole `exbild` API - and be just not starter-friendly. 
And most importantly, this all needs to work with possible custom injects into `builder` - which all is much too complicated for the small goal i had.

### PR DESCRIPTION
**Pull request by jkowalleck**
**What's the problem this PR addresses?**

resolves #6211

...

**How did you fix it?**

- added a boolean CLI switch to `builder build {bundle,plugin}` called `--metafile`, defaults to `false` 
- if CLI switch 
  - is `false`,  then the `esbuild` in instructed to NOT emit `metafile`
  - is `true`, then ...
    - the `esbuild` in instructed to emit `metafile`
    - the `metafile` data is written to target file `bundles/${name}.meta.json`
    - the target file is announced in the summary  
      example ourput:   
      ```shellSession
      $ builder build plugin --metafile 
      ➤ YN0000: ┌ Building @yarnpkg/plugin-cyclonedx
      ➤ YN0000: └ Completed in 8s 619ms
      
      ➤ YN0000: ✓ Done building @yarnpkg/plugin-cyclonedx!
      ➤ YN0000: ? Bundle path: /.../cyclonedx-node-yarn/bundles/@yarnpkg/plugin-cyclonedx.js
      ➤ YN0000: ? Bundle size: 771.52 KiB
      ➤ YN0000: ? Bundle meta: /.../cyclonedx-node-yarn/bundles/@yarnpkg/plugin-cyclonedx.meta.json
      ```
 

...

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/yarnpkg-builder/sources/commands/build/plugin.ts [hunk 1]
```diff
@@ -54,6 +54,10 @@ export default class BuildPluginCommand extends Command {
     description: `Includes a source map in the bundle`,
   });
 
+  metafile = Option.Boolean(`--metafile`, false, {
+    description: `Emit a metafile in the bundle`,
```
Comment (merceyz) [inline packages/yarnpkg-builder/sources/commands/build/plugin.ts:58]:
```suggestion
    description: `Emit a metafile next to the bundle`,
```

**Review by merceyz — COMMENTED** [review]
Would you be willing to add it to the `bundle` command as well?

File: packages/yarnpkg-builder/sources/commands/build/plugin.ts (hunk 1 above)
Comment (jkowalleck) [reply packages/yarnpkg-builder/sources/commands/build/plugin.ts]: fixed via ce4973c6e91d605adf720ac23f5d860867725f69

**Comment by jkowalleck** [comment]
> Would you be willing to add it to the `bundle` command as well?

sure. will push the needed changes in a minute

**Comment by merceyz** [comment]
Fixed the version it was complaining about, the Windows CI has been broken for a while and we haven't gotten around to fixing it yet so you can ignore that.

**Pull request merged**